protocol/message_validator.go

```diff
@@ -209,5 +209,22 @@ func ValidateReceivedChatMessage(message *protobuf.ChatMessage, whisperTimestamp
 		}
 	}
 
+	if message.ContentType == protobuf.ChatMessage_AUDIO {
+		if message.Payload == nil {
+			return errors.New("no audio content")
+		}
+		audio := message.GetAudio()
+		if audio == nil {
+			return errors.New("no audio content")
+		}
+		if len(audio.Payload) == 0 {
+			return errors.New("audio payload empty")
+		}
+
+		if audio.Type == protobuf.AudioMessage_UNKNOWN_AUDIO_TYPE {
+			return errors.New("audio type unknown")
+		}
+	}
```

Comment:
This looks like it should be a switch statement on `message.ContentType` with dedicated type validators or errors for

- `protobuf.ChatMessage_UNKNOWN_CONTENT_TYPE`
- `protobuf.ChatMessage_TRANSACTION_COMMAND`
- `protobuf.ChatMessage_STICKER`
- `protobuf.ChatMessage_IMAGE`
- `protobuf.ChatMessage_AUDIO`

Reply:
if you're going that far then you might as well make the handler of each type it's own function to keep the `switch` concise`.